### DISCUSSION TITLE
Item Suggestion: 1/4 panels

### DISCUSSION THREAD

**Discussion by omegawott** [BEEmod/BEE2-items#3789] [Feature Suggestions]
Hello, I’m a person that, as you’re about to see, likes panels too much.

I’d like to suggest a variation of panels that I’m not sure would be possible or would work with making puzzles, so apologies if I’m being ignorant here: 1/4 panels. Essentially a tiny panel that would take up 1/4 of a block/space. Think like the panels you see pull back together after GLaDOS wakes and regains control of the facility. You could fit up to four of them on a single block/space, and they’d all work independently. You could make them standard folding panels for fancy doors, or rising platforms. Ultimately this would be cool for aesthetics, as well as fun puzzle elements like creating really thin bridges that can rise out of a goo pit, dangerous to walk through. Or a single thin portal surface that can move along a room.  

Just a fun thought!

**Comment by pompousspaniard** [comment]
It's most definitely a fun thought, and if it's possible I'd definitely would like to see it added, however I think there's one problem. Please bear in mind that I'm not a developer, so this is just from my limited experience using BEEMod. I'm going to try to explain this in detail nonetheless.

In the puzzle maker, each wall is made up of "voxels" (those are squares that you see, where items can be placed). Panels, for example, take up space behind the wall they're placed on. for regular panels, it actually makes sense for the full voxel to be taken, so no problem there.
But think of the laser, for example. The laser takes one quarter of the voxel, and can be placed on the edge (not the corners, unfortunately), meaning half of the panel is left fully exposed. Great, you can place some more items on it right? No. Not really. and that's the problem, acutally.

Regardless of the in-game size of the item, the puzzle maker is still divided in voxels, which makes it easier to place items, but comes with the downside that "embedded" items take up the **whole** voxel.

I'm pretty sure there's no way for BEEMod to actually get around this, so you'd only be able to place one panel per block, which kinda takes the fun out of the "quarter panel" thing.

Don't get me wrong, it _may_ be possible, but I don't think it is.

**Edit: as FrozenDragon0 correctly pointed out there's a new tiling system in the more recent versions of BEEMod that may allow for this to happen**

**Comment by FrozenDragon0** [comment]
I believe this would now be possible thanks to the new tiling system/geometry changes.

**Reply by pompousspaniard** [reply]
Oh that's right! They did update that. I'm still using 36.1, so I kinda forgot they had done that. It'll be interesting to see if this quarter panel can be done, then.
